swiftui/PasswordlessSwiftUI/PasswordlessSwiftUI/ContentView.swift

```diff
@@ -12,72 +19,83 @@ import Firebase
 /// Main view where user can login in using Email Link Authentication.
 struct ContentView: View {
   @State private var email: String = ""
-  @State private var presentSheet = false
+  @State private var isPresentingSheet = false
 
   var body: some View {
 
     NavigationView {
-      VStack(spacing: 50.0) {
+      VStack(alignment: .leading) {
         Text("Authenticate users with only their email, no password required!")
-          .frame(maxWidth: .infinity, alignment: .leading)
-        
-        Spacer()
-          .frame(height: 0.0)
-        
+          .padding(.bottom, 60)
+
         CustomStyledTextField(
-          text: $email, placeholder: "Email", symbol: "person.circle.fill"
+          text: $email, placeholder: "Email", symbolName: "person.circle.fill"
         )
 
         CustomStyledButton(title: "Send Sign In Link") {
-          sendSignInLink(to: email)
+          sendSignInLink()
         }
+        .disabled(email.isEmpty)
 
         Spacer()
       }
       .padding()
       .navigationBarTitle("Passwordless Login")
-
     }
     .onOpenURL { (url) in
       let link = url.absoluteString
       if Auth.auth().isSignIn(withEmailLink: link) {
-        passwordlessSignIn(email: email, link: link) { (success) in
-          self.presentSheet = success
+        passwordlessSignIn(email: email, link: link) { user, error in
+          isPresentingSheet = user?.isEmailVerified ?? false
         }
       }
     }
-    .sheet(isPresented: $presentSheet) {
+    .sheet(isPresented: $isPresentingSheet) {
       SuccessView(email: $email)
     }
+  }
```

Comment:
Just thought of a potential race condition here: if someone sends a sign in link, changes the email in the text field, then goes to their email and logs in, we'll show that they're logged in as the content of the text field, not the email address they actually used.

Can you think of some ways to get around this without changing the UI?

Reply:
Update. When this code is triggered on a URL open, the `email` passed into `passwordlessSignIn` will be the current email typed in the textfield. `passwordlessSignIn` really just wraps Auth's "sign in with email and link method", which matches up the link and email and will return an error if the email and email the link was sent to don't match. 

So I dont think this race condition will happen because if a user enters email A and sends the link. Then changes the textfield to email B and then goes to open the link sent to email A. They will not be signed in when returning to the app because the link corresponding with email A no longer corresponds to the binded `email` property, which is now email B.

Reply:
One could always keep the latest email the link was sent to when a user taps the "send sign in link" button. But then what if they send the link to 5 emails and only open the link on the 1,2,3, or 4th? So I think this might be ok to leave as it with the idea that whatever email is in the textfield should represent the email the users wants to login in with.